devices/At24cxx/At24Base.cs

```diff
@@ -0,0 +1,185 @@
+        /// <param name="pageSize">The device page size in bytes.</param>
+        /// <param name="pageCount">The number of pages on the device.</param>
+        public At24Base(I2cDevice i2cDevice, ushort pageSize, ushort pageCount)
+        {
+            if (i2cDevice == null)
+            {
+                throw new ArgumentNullException(nameof(i2cDevice));
+            }
+
+            _i2cDevice = i2cDevice;
+            PageSize = pageSize;
+            PageCount = pageCount;
+            Size = pageSize * pageCount;
+        }
+
+        /// <summary>
+        /// Reads a single byte from the device at the address following the last byte read or written.
+        /// </summary>
+        /// <returns>The byte read from the device.</returns>
+        public byte ReadByte()
+        {
+            return _i2cDevice.ReadByte();
+        }
+
+        /// <summary>
+        /// Reads a single byte from the device at the given address.
+        /// </summary>
+        /// <param name="address">The address to read from.</param>
+        /// <returns>The byte read from the device.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when address falls outside of the accessible range for this device.</exception>
+        public byte ReadByte(int address)
+        {
+            VerifyMemoryAddress(address, Size);
+
+            byte[] readBuffer = new byte[1];
+            byte[] writeBuffer = CreateWriteBuffer(address, 0);
+
+            _i2cDevice.WriteRead(writeBuffer, readBuffer);
+
+            return readBuffer[0];
+        }
+
+        /// <summary>
+        /// Reads multiple bytes from the device starting from the given address.
+        /// </summary>
+        /// <param name="address">The address to read from.</param>
+        /// <param name="length">The number of bytes to read.</param>
+        /// <returns>The bytes read from the device.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when address falls outside of the accessible range for this device.</exception>
+        public byte[] Read(int address, int length)
+        {
+            VerifyMemoryAddress(address, Size);
+
+            byte[] readBuffer = new byte[length];
+            byte[] writeBuffer = CreateWriteBuffer(address, 0);
+
+            _i2cDevice.WriteRead(writeBuffer, readBuffer);
+
+            return readBuffer;
+        }
+
+        /// <summary>
+        /// Writes a single byte to the device at the given address.
+        /// </summary>
+        /// <param name="address">The address to write to.</param>
+        /// <param name="value">The byte to be written into the device.</param>
+        /// <returns>The number of bytes successfully written to the device.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">Thrown when address falls outside of the accessible range for this device.</exception>
+        public uint WriteByte(int address, byte value)
```

Comment:
and same question for address here.

Reply:
any max address to check if it is valid?

Reply:
I moved the address validation into the CreateWriteBuffer function. Since CreateWriteBuffer is called by all of the Read and Write functions to tack the memory address onto the data being sent to the chip, it also seemed like a good place to verify the address itself. The only function that required some extra validation is Read, since the length of data being read might run past the end of the max address.

Reply:
I moved the address validation into the CreateWriteBuffer function. Since CreateWriteBuffer is called by all of the Read and Write functions to tack the memory address onto the data being sent to the chip, it also seemed like a good place to verify the address itself. The only function that required some extra validation is Read, since the length of data being read might run past the end of the max address. (*whoops, moved this comment, I think I put this reply in the wrong spot)